### PR TITLE
[L0] fix waiting on non-owned native handle events

### PR DESCRIPTION
**Pull request by pbalcer**
UR testing.

Depends on https://github.com/intel/llvm/pull/12375

### CODE REVIEW

**Comment by kbenzie** [comment]
Looks like the Windows failure is the same as we're seeing on https://github.com/intel/llvm/pull/12375.

**Comment by pbalcer** [comment]
> Looks like the Windows failure is the same as we're seeing on #12375.

Yes, I'm waiting for your fix to get merged.

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -56,14 +56,8 @@ endif()
...
-  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
-  # commit 79c28d0f0713f58358d5080653d95803fd131749
-  # Merge: 25e0b603 45d76b78
-  # Author: aarongreig <aaron.greig@codeplay.com>
-  # Date: Fri Jan 12 16:14:44 2024 +0000
-  #     Merge pull request #1186 from hdelan/device-global-hip
-  #     [HIP] Add support for global variable read write
-  set(UNIFIED_RUNTIME_TAG 79c28d0f0713f58358d5080653d95803fd131749)
+  set(UNIFIED_RUNTIME_REPO "https://github.com/pbalcer/unified-runtime.git")
+  set(UNIFIED_RUNTIME_TAG 3ef6d11d45a3abb9bb3f81deed19bbb69ffd33a1)
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:60]:
```suggestion
  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
  # commit 92f44da347323af842cb5352b94169d16a5cfbe7
  # Merge: 43c2ad23 3ef6d11d
  # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
  # Date:   Fri Jan 19 15:59:50 2024 +0000
  #     Merge pull request #1257 from pbalcer/fix-native-handle-waits
  #     [L0] fix waiting on non-owned native handle events
  set(UNIFIED_RUNTIME_TAG 92f44da347323af842cb5352b94169d16a5cfbe7)
```

**Review by kbenzie — COMMENTED** [review]
https://github.com/oneapi-src/unified-runtime/pull/1257 has merge so the tag can be updated.

**Comment by kbenzie** [comment]
Pulling in tip sycl branch once https://github.com/intel/llvm/pull/12375 has merged should fix the Windows failure.

**Comment by kbenzie** [comment]
@intel/llvm-gatekeepers please merge

**Pull request merged**